assets/src/editor/events/hocs/with-editor-event-entity.js

```diff
@@ -0,0 +1,23 @@
+import { withSelect } from '@wordpress/data';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { isModelEntityOfModel } from '@eventespresso/validators';
+
+const DEFAULT_OBJECT = {
+	eventEntity: null,
+};
+
+export default createHigherOrderComponent(
+	withSelect( ( select, { eventEntity } ) => {
+		// if we already have an event entity, just pass it through
+		if ( isModelEntityOfModel( eventEntity, 'event' ) ) {
+			return { eventEntity };
+		}
+		const events = select( 'eventespresso/core' ).getEvents();
+		return events.length > 0 ?
+			{
+				eventEntity: events.slice( -1 ).pop(),
+			} :
+			DEFAULT_OBJECT;
+	} ),
+	'withEditorEventEntity'
```

Comment:
Think of this replacing the `eventEntity` prop being passed down through from the top-level component.  This hoc retrieves the event entity from the store state.  Note, I did introduce some logic here so that if there is already an `eventEntity` being passed through as props on a component that is composed with this hoc, then that `eventEntity` will take priority.  It basically prevents an unnecessary query to the store state (even though that won't be expensive).